astroplan/constraints.py

```diff
@@ -394,7 +395,81 @@ def compute_constraint(self, times, observer, targets):
                              "MoonSeparationConstraint.")
         return mask
 
+    
+class DarkGreyBrightConstraint(Constraint):
+    """
+    Constrain whether it is dark, gray or bright time or how bright.
+    """
+    # TODO: make a new test
+    @u.quantity_input(horizon=u.deg)
+    def __init__(self, lunar_horizon=90*u.deg, min_moon_illumination=None,
+                                                max_moon_illumination=None):
+        '''
+        Parameters
+        ----------
+        Lunar_horizon : `~astropy.units.Quantity` (optional)
+            The altitude of the ``horizon`` and defaults to 90 degrees       
```

Comment:
Just to chime in, I can easily foresee a use case for light-pollution
levels. Not sure if this is the appropriate place for that but it seems to
be in line with this Constraint without involving the moon at all.

On Wed, Jun 15, 2016 at 9:20 AM, StuartLittlefair <notifications@github.com>
wrote:

> In astroplan/constraints.py
> <https://github.com/astropy/astroplan/pull/174#discussion_r67182909>:
>
> > @@ -394,7 +395,81 @@ def compute_constraint(self, times, observer, targets):
> >                               "MoonSeparationConstraint.")
> >          return mask
> >
> > +
> > +class DarkGreyBrightConstraint(Constraint):
> > +    """
> > +    Constrain whether it is dark, gray or bright time or how bright.
> > +    """
> > +    # TODO: make a new test
> > +    @u.quantity_input(horizon=u.deg)
> > +    def __init__(self, lunar_horizon=90*u.deg, min_moon_illumination=None,
> > +                                                max_moon_illumination=None):
> > +        '''
> > +        Parameters
> > +        ----------
> > +        Lunar_horizon : `~astropy.units.Quantity` (optional)
> > +            The altitude of the ``horizon`` and defaults to 90 degrees
>
> The only issue with this is that most astronomers have a different
> conception of what dark, grey and bright mean. Dark time is moon
> illuminations < 0.25, even if the moon is up.
>
> And then, since this is always going to be used as a constraint on the
> brightness of the night sky, it doesn't make sense to assume the moon is
> up, since the night sky can be darker when the Moon is 65% illuminated, but
> below the horizon, than it is when the moon is up but 30% illuminated.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/astropy/astroplan/pull/174/files/8a8fcaebad00584066c950ea2c8161ce21d35d7e#r67182909>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAEUUM4YWUnV3x9I1o3iLZgvnvHzGKCaks5qMBhYgaJpZM4I1rL5>
> .
>



-- 
~Wilfred Tyler Gee